### PR TITLE
Tests showing user defined aggregate returning a struct

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/issues/600

 # Rationale for this change
It turns out that DataFusion supports user defined aggregates returning a struct but there is very little end to end SQL testing of the feature

Adding end to end testing will ensure that the feature doesn't get broken accidentally any maybe provides an example for others to start from when working on their own features

# What changes are included in this PR?
Add a new "rust integration test" showing a user defined aggregate returning a struct. It can be run like

```shell
cargo test -p datafusion --test user_defined_aggregates
```

# Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/tests/sql/mod.rs [hunk 1]
```diff
@@ -108,12 +108,12 @@ pub mod window;
 
 pub mod arrow_typeof;
 pub mod decimal;
-mod explain;
-mod idenfifers;
+pub mod explain;
```
Comment (alamb) [inline datafusion/core/tests/sql/mod.rs:111]: This is a driveby cleanup to make module definitions in `sql/mod.rs` consistent

File: datafusion/core/tests/user_defined_aggregates.rs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        StateTypeFunction, TypeSignature, Volatility,
+    },
+    physical_plan::Accumulator,
+    prelude::SessionContext,
+    scalar::ScalarValue,
+};
+
+#[tokio::test]
+/// Basic query for with a udaf returning a structure
+async fn test_udf_returning_struct() {
```
Comment (alamb) [inline datafusion/core/tests/user_defined_aggregates.rs:44]: It is actually quite cool that this works 👌

**Review by andygrove — APPROVED** [review]

**Review by yjshen — APPROVED** [review]
